tangelo/linq/target/target_qulacs.py

```diff
@@ -75,6 +75,25 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
                 else:
                     state.set_zero_state()
             python_statevector = None
+        elif save_mid_circuit_meas:
+            n_meas = source_circuit._gate_counts.get("MEASURE", 0)
+            samples = list()
+            full_samples = list()
+            python_statevector = None
+            for _ in range(self.n_shots):
+                translated_circuit.update_quantum_state(state)
+                join = "".join([str(state.get_classical_value(i)) for i in range(n_meas)])
+                measurement = join
+                sample = self._int_to_binstr(state.sampling(1)[0], source_circuit.width)
+                full_samples += [measurement + sample]
```

Comment:
Does this become a huge list for large values of n_shots ?
I would rather this be a dictionary, to which we increment values, rather than a big list that is then turned into a dict by Counter.

The built-in `get` method of dictionaries would make this pretty straightforward. Assume `measurement` and `sample` are bit strings (e.g `01101`)
```
all_samples[measurement+sample] = all_samples.get(measurement+sample, 0) + 1
```

This can then be normalized to obtained frequencies.

This part of the code may have nothing to do with your actual backend choice: we may change all built-in backends, or consider providing functions in the base class or a toolbox to automatically make this available to all backends in a generic way. I am ok with either. But ideally, I want to avoid duplicating code across backends when it is not necessary.

Reply:
Good point. Similar in the previous code block. I think it was done to stay consistent with the `qulacs` sampling method which returns a list of samples. I simplified these two, but the third block probably has to stay as it is.